pkg/controllers/provisioningv2/cluster/remove.go

```diff
@@ -57,8 +43,6 @@ func (h *handler) updateClusterStatus(cluster *v1.Cluster, status v1.ClusterStat
 	if equality.Semantic.DeepEqual(status, cluster.Status) {
 		return cluster, previousErr
 	}
-	cluster = cluster.DeepCopy()
-	cluster.Status = status
 	cluster, err := h.clusters.UpdateStatus(cluster)
```

Comment:
I changed the logic. Previously, the changes to the cluster's status were made to `status`. So, if a change was detected, then the `cluster.Status` needed to be updated for the `UpdateStatus` call.

Now, the changes are made to `cluster.Status` and `status` is the "old" status. So, we don't need to update `cluster.Status` because it is the new status.

Reply:
OK I see what you're doing here. 

What I meant by this, was is it necessary to have an independent function like `updateClusterStatus` instead of just directly calling `h.clusters.UpdateStatus(cluster)` directly if the `UpdateStatus` does not push an update based based on the diff of the objects (so basically, if it also does the equality.Semantic.DeepEqual`

this function confuses me because the passed in `status v1.ClusterStatus` may be different than the `cluster.Status` which would call a `h.clusters.UpdateStatus(cluster)` so it relies on the caller basically call it like `updateClusterStatus(cluster, cluster.status, someErr)`

Reply:
I think we can check the server and see if we need to update the status (which I thought `UpdateStatus` would do)

Right now the way this is written the caller has to store a copy of the old status to pass in to see if the status needs to be updated. I think it's a strange/confusing abstraction